File: 1635-number-of-good-pairs/1635-number-of-good-pairs.py
from collections import defaultdict
class Solution:
    def fact(self,num):
        prod=0
        while(num>0):
            prod+=num
            num-=1
        return prod
    def numIdenticalPairs(self, nums: List[int]) -> int:
        m=defaultdict(list)
        for i in range(len(nums)):
            m[nums[i]].append(i)
        s=0
        for i in m.values():
            s+=self.fact(len(i)-1)
        return s

File: 1635-number-of-good-pairs/1635-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        map<int,vector<int>>hash;
        for(int i=0;i<nums.size();i++){
            hash[nums[i]].push_back(i);
        }
        int count=0;
        for(auto i:hash){
            count+=(i.second.size())*(i.second.size()-1)/2;
        }
        return count;
    }
};